hobbify/habits/views/habits.py

```diff
@@ -0,0 +1,209 @@
+                status=status.HTTP_406_NOT_ACCEPTABLE
+            )
+
+        instance = RecurrentInstance.objects.create(
+            habit=habit,
+            done=False,
+            rule=str(rule),
+            recurrence=str(todays_intance)
+        )
+        data = RecurrenceModelSerializer(instance).data
+
+        return Response(
+            data,
+            status=status.HTTP_201_CREATED
+        )
+
+    @action(detail=True, methods=['get'])
+    def today(self, request, *args, **kwargs):
+        """ Return today's instance of an habit """
+        habit_id = self.kwargs['id']
+        rule = self.set_rule(habit_id)
+        today = datetime.datetime.now().strftime("%Y-%m-%d")
+
+        instances = RecurrentInstance.objects.filter(is_hidden=False, habit=habit_id)
+        instances_data = [RecurrenceModelSerializer(instance).data for instance in instances]
+        todays_intance = {}
+
+        for info in instances_data:
+            instance_date = str(info['recurrence']).partition(' ')[0]
+            print(instance_date, today)
+            if instance_date == today and str(info['rule']) == str(rule):
+                todays_intance = info
+
+        return Response(todays_intance, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=['get'])
+    def all(self, request, *args, **kwargs):
+        """ Return all recurrences of a habit """
+        rule = self.set_rule(self.kwargs['id'])
+        temp = [{"date": s} for s in rule]
+
+        data = {
+            "rule": str(rule),
+            "instances": temp
+        }
+
+        return Response(data, status=status.HTTP_200_OK)
+
+    def get_today_instance(self, id):
+        """ Return today's recurrent instance """
+        rule = self.set_rule(id)
+        today = datetime.datetime.now().strftime("%Y-%m-%d")
+        todays_intance = ""
+
+        for s in rule:
+            instance_date = str(s).partition(' ')[0]
+            if instance_date == today:
+                todays_intance = s
+                break
+
+        return todays_intance
+
+    def set_rule(self, habit_id):
+        """ Set a rrule based on database habit information """
+        habit = Habit.objects.filter(id=habit_id).first()
+
+        start = habit.start_date
+        freq = DAILY
+        inter = 1
+        end = habit.end_date if habit.end_date is not None else habit.start_date + datetime.timedelta(days=21)
+
+        if habit.frequency == 1:
+            freq = DAILY
+        elif habit.frequency == 2:
+            freq = WEEKLY
+        elif habit.frequency == 3:
+            freq = MONTHLY
+        elif habit.frequency == 4:
+            freq = WEEKLY
+            inter = 3
+        else:
+            freq = WEEKLY
+            inter = 5
+
+        habit_recurrences_rule = rrule(
+            freq,
+            dtstart=start,
+            until=end,
+            interval=inter,
+            cache=True
+        )
+
+        return habit_recurrences_rule
```

Comment:
This computation must happen just once (when modifying or during first creation). Do not re-compute this every time.